test/e2e/cert_manager_deployment_test.go

```diff
@@ -237,47 +237,67 @@ func TestCertRenew(t *testing.T) {
 		},
 	}
 	certmanagerv1Client, err := certmanagerclientset.NewForConfig(config)
-	require.NoError(t, err)
+	require.NoErrorf(t, err, "failed to create cert manager client: %v", err)
 	crt, err = certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Create(ctx, crt, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create certificate: %v", err)
 	defer certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Delete(ctx, crt.ObjectMeta.Name, metav1.DeleteOptions{})
-	require.NoError(t, err)
-	err = wait.PollImmediate(PollInterval, TestTimeout, func() (bool, error) {
+
+	err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 20*time.Second, true, func(ctx context.Context) (bool, error) {
+		crt, err = certmanagerv1Client.CertmanagerV1().Certificates(crt.ObjectMeta.Namespace).Get(ctx, crt.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Errorf("failed to fetch certificate after creation: %v", err)
+			return false, nil
+		}
+		for _, condition := range crt.Status.Conditions {
+			if condition.Type == certmanagerv1.CertificateConditionReady && condition.Status == certmanagermetav1.ConditionTrue {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
```

Comment:
Maybe we could replace these with the utils func here: https://github.com/openshift/cert-manager-operator/blob/67689422c30cd0b487966666542df3afef791042/test/e2e/utils_test.go#L334

Reply:
Agree, suggestion incorporated.

Reply:
@lunarwhite ^^